cypress/support/commands.js

```diff
@@ -10,7 +10,28 @@
 //
 //
 // -- This is a parent command --
-// Cypress.Commands.add("login", (email, password) => { ... })
+Cypress.Commands.add("login", (username, password) => {
+    cy.visit(Cypress.env('login_url'));
+        cy.get("[name=csrfmiddlewaretoken]")
+            .should("exist")
+            .should("have.attr", "value")
+            .as("csrfToken");
+
+        cy.get("@csrfToken").then((token) => {
+            cy.request({
+                method: "POST",
+                url: Cypress.env("login_url"),
+                form: true,
+                body: {
+                    username: username,
+                    password: password,
+                },
+                headers: {
+                    "X-CSRFTOKEN": token,
+                },
+            });
```

Comment:
I ran into some trouble with [some very similar code](https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/blob/master/cypress/integration/admin.spec.js#L29-L33): if the POST fails for some reasons, like an invalid password, then the `login()` method still succeeds, and debugging the failing parts of the test after `login()` gets very confusing, because the user is not logged in.

Reply:
We fixed this on one of the CI branches so it'll show up here soon!